components/application-authenticators/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -127,11 +132,66 @@ protected String getAuthenticateUser(OAuthClientResponse token) {
     }
 
     /**
-     * @param token
-     * @return
+     * Get user info endpoint.
+     * @param token OAuthClientResponse
+     * @param authenticatorProperties Map<String, String>
+     * @return User info endpoint.
      */
-    protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse token) {
-        return new HashMap<ClaimMapping, String>();
+    protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        return authenticatorProperties.get(IdentityApplicationConstants.Authenticator.OIDC.USER_INFO_URL);
+    }
+
+    /**
+     * Get subject attributes.
+     * @param token OAuthClientResponse
+     * @param authenticatorProperties Map<String, String>
+     * @return Map<ClaimMapping, String> Claim mappings.
+     */
+    protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse token,
+                                                             Map<String, String> authenticatorProperties) {
+
+        Map<ClaimMapping, String> claims = new HashMap<ClaimMapping, String>();
+
+        try {
+
+            String accessToken = token.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            String url = getUserInfoEndpoint(token, authenticatorProperties);
+
+            String json = sendRequest(url, accessToken);
+
+            if (!StringUtils.isNotBlank(json)) {
+                log.info("Unable to fetch user claims. Proceeding without user claims");
+                return claims;
+            }
+
+            Map<String, Object> jsonObject = JSONUtils.parseJSON(json);
+
+            // Extract the inside profile JSON object.
+            Map<String, Object> profile = JSONUtils.parseJSON(
+                    jsonObject.entrySet().iterator().next().getValue().toString());
+
+            if (profile == null) {
+                log.info("Invalid user profile object. Proceeding without user claims");
+                return claims;
+            }
+
+            for (Map.Entry<String, Object> data : profile.entrySet()) {
+
+                String key = data.getKey();
+
+                claims.put(ClaimMapping.build(key, key, null, false), profile.get(key).toString());
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Adding claims from end-point data mapping : " + key + " - " +
+                            profile.get(key).toString());
+                }
+            }
+
+        } catch (Exception e) {
```

Comment:
Any reason for catching Exception rather than handling different types of exceptions separately ?